### DISCUSSION TITLE
Help with Debugging JavaScript Event Listener Issue

### DISCUSSION THREAD

**Discussion by theidioms** [community/community#132900] [Programming Help] [answered]
Hi everyone,

I'm relatively new to JavaScript and have encountered a problem that I can't seem to figure out. I'm working on a simple project where I need to change the text of a button when it is clicked. Here's the code snippet I'm working with:

``
document.getElementById('myButton').addEventListener('click', function() {
    this.textContent = 'Clicked!';
});
``

For some reason, when I click the button, nothing happens. I've double-checked the HTML, and the button has the correct ID. Here's the relevant HTML:

```
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Button Click Example</title>
</head>
<body>
    <button id="myButton">Click me</button>

    <script src="script.js"></script>
</body>
</html>
```

I've tried adding console.log statements within the event listener, and they don't seem to be firing either. What am I missing here?

Any help would be greatly appreciated!
Thanks and Regards,
Elena [Idioms](https://www.theidioms.com/)

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by davevad93** [comment, marked as answer]
Hi @theidioms ,

Have you added the `document.addEventListener` in your `js` script? Try this:

```js
document.addEventListener('DOMContentLoaded', function() {
    document.getElementById('myButton').addEventListener('click', function() {
        this.textContent = 'Clicked!';
    });
});
```
Before clicking:

![0](https://github.com/user-attachments/assets/5e52eae8-4fc2-43ef-ab8c-a89d87d41a20)

After clicking:

![1](https://github.com/user-attachments/assets/ef0bf0b3-0e44-401b-b6c0-c37731581e5c)

**Reply by theidioms** [reply]
@davevad93 
Thanks for the reply. The problem is resolved now.

Elena

**Reply by davevad93** [reply]
You're welcome @theidioms , have a nice day!
